### PR TITLE
NSPR 6035 clock throttling enhancements

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Added changes in XRT driver to report clock throttling parameters through xbmgmt tool.

Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
**On U2 device**
# xbmgmt config --show
Daemon:
	host=amit-PowerEdge-T640
0000:3e:00.0:
	security level: 0
	Runtime clock scaling feature:
		Supported: Yes
		Enabled: Yes
		Threshold limits:
			Power: 29W
			Temperature: 100C
		Power threshold override:
 			Override: Enabled
			Override limit: 29W
		Temperature threshold override:
 			Override: Enabled
			Override limit: 100C
	Data Retention: Disable

**On U50 device**
# xbmgmt config --show
Daemon:
	host=xsjsda160
0000:3b:00.0:
	security level: 0
	Runtime clock scaling feature:
		Supported: Yes
		Enabled: Yes
		Threshold limits:
			Power: 72W
			Temperature: 92C
		Power threshold override:
 			Override: Disabled
			Override limit: 0W
		Temperature threshold override:
 			Override: Disabled
			Override limit: 0C
	Data Retention: Disable

**Comment by rajkumar-xilinx** [comment]
> Can you add these changes to xbmgmt2 as well? If you have any questions about how to add the changes to the new arch, please schedule a meeting

Hi @AShivangi , Thanks for reviewing the change. I am going through xbmgmt2 now. Is it fine to prepare a new pull request for xbmgmt2 changes?
Let this change go into master and I will add all necessary changes in xbmgmt2 in the next pull request.

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_config.cpp [hunk 1]
```diff
@@ -28,13 +28,13 @@
...
 const char *subCmdConfigUsage =
-    "--device [--card bdf] --runtime_clk_scale en(dis)able\n"
+    "--device [--card bdf] [--runtime_clk_scale (en/dis)able] [--cs_threshold_power_override val] [--cs_threshold_temp_override val] [--cs_reset val]\n"
     "--enable_retention [--ddr] [--card bdf]\n"
     "--disable_retention [--ddr] [--card bdf]\n";
 const char *subCmdConfigExpUsage =
     "Experts only:\n"
     "--daemon --host ip-or-hostname-for-peer\n"
-    "--device [--card bdf] [--security level] [--cs_threshold_power_override val] [--cs_reset val]\n"
+    "--device [--card bdf] [--security level] [--cs_threshold_power_override val] [--cs_threshold_temp_override val] [--cs_reset val]\n"
```
Comment (AShivangi) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_config.cpp:37]: if an option is already displayed to the user, it doesn't make sense to add it to experts only command as well. Please remove `[--cs_threshold_power_override val] [--cs_threshold_temp_override val] [--cs_reset val]`

Has this been signed off by PRT? I remember that these options got moved to experts only for some reason. Maybe Max can confirm

**Comment by AShivangi** [comment]
> > Can you add these changes to xbmgmt2 as well? If you have any questions about how to add the changes to the new arch, please schedule a meeting
> 
> Hi @AShivangi , Thanks for reviewing the change. I am going through xbmgmt2 now. Is it fine to prepare a new pull request for xbmgmt2 changes?
> Let this change go into master and I will add all necessary changes in xbmgmt2 in the next pull request.

Yes, that sounds good to me. I added a comment about the help menu. Other than that the changes look good.

**Comment by rajkumar-xilinx** [comment]
I have update config help menu as per @AShivangi comments.

xbmgmt config --help
'config' sub-command usage:
--device [--card bdf] [--runtime_clk_scale enable|disable]
--enable_retention [--ddr] [--card bdf]
--disable_retention [--ddr] [--card bdf]

**Pull request merged**